src/RestrictedTokenFaucet.t.sol

```diff
@@ -0,0 +1,160 @@
+pragma solidity ^0.5.4;
+
+import "ds-test/test.sol";
+import "ds-token/token.sol";
+
+import "./RestrictedTokenFaucet.sol";
+
+contract FaucetUser {
+    DSToken token;
+    RestrictedTokenFaucet faucet;
+
+    constructor(DSToken token_, RestrictedTokenFaucet faucet_) public {
+        token = token_;
+        faucet = faucet_;
+    }
+
+    function doGulp() public {
+        faucet.gulp(address(token));
+    }
+
+    function doBalanceOf() public view returns(uint) {
+        return token.balanceOf(address(this));
+    }
+
+    function doUnDone(address usr) public {
+        faucet.unDone(usr, address(token));
+    }
+
+    function doRely(address usr) public {
+        faucet.rely(usr);
+    }
+
+    function doDeny(address usr) public {
+        faucet.deny(usr);
+    }
+
+    function doShut() public {
+        faucet.shut(ERC20Like(address(this)));
+    }
+
+    function doSetAmt(uint amt) public {
+        faucet.setamt(amt);
+    }
+}
+
+contract RestrictedTokenFaucetTest is DSTest {
+    RestrictedTokenFaucet faucet;
+    DSToken token;
+    address user1;
+    address user2;
+    address self;
+
+    function setUp() public {
+        faucet = new RestrictedTokenFaucet(20 ether);
+        token = new DSToken("TEST");
+        token.mint(address(faucet), 1000000 ether);
```

Comment:
I find the usage of ether here confusing, given this is a DSToken and not Eth. Maybe just use simple uints?

```suggestion
        token.mint(address(faucet), 100);
```

Then I would maybe add another test with Eth - to make sure it works with both native Eth and ERC20's.

Reply:
I'll update it to remove the confusion, but the faucet isn't meant to work with ETH only ERC-20s